### PR TITLE
✅(frontend) clear warning from useEnroll test

### PR DESCRIPTION
**Pull request by rlecellier**
one act warning and some unmocked requests

## Purpose

Keep warning free tests

### CODE REVIEW

**Comment by NathanVss** [comment]
Unfortunately the `test-front` job fails, I will do the review once this is fixed 🙏

File: src/frontend/js/hooks/useEnroll/index.spec.tsx [hunk 1]
```diff
@@ -39,17 +39,24 @@ describe('useEnroll ( edge case )', () => {
...
   });
 
   it('should return early when a course run is not enrollable', async () => {
+    fetchMock.get('https://joanie.endpoint/api/v1.0/enrollments/', []);
     fetchMock.post('https://joanie.endpoint/api/v1.0/enrollments/', {});
     const { result } = renderHook(() => useEnroll([], undefined), { wrapper });
-    await waitFor(() => {
-      expect(result.current?.enroll).toBeDefined();
-    });
+    await waitFor(() => expect(result.current.isLoading).toBeFalsy());
```
Comment (jbpenrath) [inline src/frontend/js/hooks/useEnroll/index.spec.tsx:58]: Stricter is better. Prefer `toBe` to prevent false positive tests.
```suggestion
    await waitFor(() => expect(result.current.isLoading).toBe(false));
```

File: src/frontend/js/hooks/useEnroll/index.spec.tsx [hunk 2]
```diff
@@ -60,13 +67,19 @@ describe('useEnroll ( edge case )', () => {
...
   it('should not return early when a course run is enrollable', async () => {
+    fetchMock.get('https://joanie.endpoint/api/v1.0/enrollments/', []);
     fetchMock.post('https://joanie.endpoint/api/v1.0/enrollments/', {});
-    const { result } = renderHook(() => useEnroll([], undefined), { wrapper });
-    await waitFor(() => {
-      expect(result.current?.enroll).toBeDefined();
+    const { result } = renderHook(() => useEnroll([], undefined), {
+      wrapper,
     });
+    await waitFor(() => expect(result.current.isLoading).toBeFalsy());
```
Comment (jbpenrath) [inline src/frontend/js/hooks/useEnroll/index.spec.tsx:75]: Ditto
```suggestion
    await waitFor(() => expect(result.current.isLoading).toBe(false));
```

File: src/frontend/js/hooks/useEnroll/index.spec.tsx [hunk 3]
```diff
@@ -60,13 +67,19 @@ describe('useEnroll ( edge case )', () => {
...
+
     const courseRun = JoanieCourseRunFactory().generate();
-    await result.current.enroll(courseRun);
+
+    act(() => {
+      result.current.enroll(courseRun);
+    });
+    await waitFor(() => expect(result.current.isLoading).toBeFalsy());
```
Comment (jbpenrath) [inline src/frontend/js/hooks/useEnroll/index.spec.tsx:82]: Ditto
```suggestion
    await waitFor(() => expect(result.current.isLoading).toBe(false));
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -18,6 +18,10 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 - Prevent internal server error when course runs sync webhook targets
```
Comment (NathanVss) [inline CHANGELOG.md:18]: IMO the commit body "one act warning and some unmocked requests" must focus more on the why and could be more self explanatory like "some tests were emitting warnings due to act misusage and unmocked requests, so we decided to fix those".

File: CHANGELOG.md [hunk 5]
```diff
@@ -13,6 +13,10 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 - Added DashboardItemOrder that allows to select course runs
 - Add make dbshell cmd to access database in cli
 
+## Changes
+
+- clear useEnroll test warning
```
[the remark this answers is not in the document]
Comment (rlecellier) [reply CHANGELOG.md:23]: Who would we do it ? 
We can: 
* ignore the red CI and merge.
* try to ignore `.spec` files from the CI changelogs check.

[the remark this answers is not in the document]
Comment (NathanVss) [reply CHANGELOG.md:23]: The changelog CI job isn't mandatory so it's ok to merge a PR if its red. It just acts like a warning.

**Review by jbpenrath — APPROVED** [review]
LGTM ✅

**Review by NathanVss — APPROVED** [review]
Let's rock
